src/components/carousel-func.js

```diff
@@ -2,18 +2,21 @@
 window.onload = function(){
 let slidePosition = 0;
 const slides = document.getElementsByClassName('carousel__item');
+const nextButton = document.getElementById('carousel__button--next');
+const prevButton = document.getElementById('carousel__button--prev');
 const totalSlides = slides.length;
 
-document.
-  getElementById('carousel__button--next')
-  .addEventListener("click", function() {
+if (nextButton) {
```

Comment:
This checks to see if they are on the page before attempting to add event listeners to them.